<?xml version="1.0" encoding="UTF-8"?>
<launch>
  <arg name="voxel_size" default="0.2"/>
  <arg name="voxels_per_side" default="16" />
  <arg name="truncation_distance" default="0.6" />
  <arg name="method" default="fast"/>
  <arg name="ray_length" default="5" />
  <arg name="min_ray_length" default="0.2" />

  <!-- defines name of the namespace of the drone -->
  <arg name="UAV_NAME" default="$(optenv UAV_NAME uav1)" />

  <arg name="world_frame" default="map" />

  <group ns="$(arg UAV_NAME)">

    <!--Voxblox Mapping-->
    <node name="voxblox_node" pkg="voxblox_ros" type="esdf_server" output="screen" args="-alsologtostderr">
      <remap from="pointcloud" to="/$(arg UAV_NAME)/pcl_filter_rs_front_pitched/points_processed"/>
      <param name="tsdf_voxel_size" value="$(arg voxel_size)" />
      <param name="tsdf_voxels_per_side" value="$(arg voxels_per_side)" />
      <param name="truncation_distance" value="$(arg truncation_distance)" />
      <param name="method" value="$(arg method)" />
      <param name="max_ray_length_m" value="$(arg ray_length)" />
      <param name="min_ray_length_m" value="$(arg min_ray_length)" />

      <param name="publish_tsdf_map" value="true" />
      <param name="publish_esdf_map" value="true" />

      <param name="publish_pointclouds" value="true" />
      <param name="publish_tsdf_info" value="true" />

      <param name="use_freespace_pointcloud" value="true" />
      <remap from="freespace_pointcloud" to="/$(arg UAV_NAME)/freespace_pc_rs_front_pitched/points_freespace"/>
      <param name="color_mode" value="normals" />
      <param name="use_tf_transforms" value="true" />
      <param name="update_mesh_every_n_sec" value="2.0" />
      <param name="min_time_between_msgs_sec" value="0.1" />
      <param name="pointcloud_queue_size" value="100" />
      <param name="use_const_weight" value="false" />

      <param name="verbose" value="true" />
      <param name="world_frame" value="$(arg world_frame)" />
    </node>

  </group>

</launch>